src/internal/utils/bucketHelper.ts

```diff
@@ -850,8 +850,9 @@ export const transformMeasureBuckets = (
 ) => {
     let unusedMeasures: IBucketItem[] = [];
 
-    const newBuckets: IBucket[] = measureBucketItemsLimits.map(({ localIdentifier, itemsLimit }) => {
-        const preferredBucketItems = getPreferredBucketItems(buckets, [localIdentifier], [METRIC]);
+    const newBuckets: IBucket[] = measureBucketItemsLimits.map(({ localIdentifiers, itemsLimit }) => {
```

Comment:
how about if we rewrite this function as below? Then all refactoring on other files can be reverted.
```javascripts
export const transformMeasureBuckets = (
    measureBucketItemsLimits: IMeasureBucketItemsLimit[],
    buckets: IBucket[],
) => {
    let unusedMeasures: IBucketItem[] = [];
    const newBuckets: IBucket[] = measureBucketItemsLimits.map(({ localIdentifier, itemsLimit }) => {
    const preferedBucketlocalIdentifiers = localIdentifier === BucketNames.MEASURES
        ? [BucketNames.MEASURES, BucketNames.SIZE]
        : localIdentifier === BucketNames. SECONDARY_MEASURES
            ? [BucketNames.SECONDARY_MEASURES, BucketNames.COLOR]
            : [localIdentifier];
    const preferredBucketItems = getPreferredBucketItems(buckets, preferedBucketlocalIdentifiers, [METRIC]);
    const measuresToBePlaced = preferredBucketItems.splice(0, itemsLimit);

    if (measuresToBePlaced.length === 0) {
        return {
            localIdentifier,
            items: unusedMeasures.splice(0, itemsLimit),
        };
    }
    unusedMeasures = [...unusedMeasures, ...preferredBucketItems];
    return {
        localIdentifier,
        items: measuresToBePlaced,
    };
});
```

Reply:
updated